### DISCUSSION TITLE
Upload local neovim systems to remote server.

### DISCUSSION THREAD

**Discussion by Saton-anger** [ayamir/nvimdots#967] [General]
Hi! Wonderful project and it really make my daily coding staff much more easier than before! However, I am wondering how can I upload the local settings to the remote linux server, because I am not one of the sudors of the server...

**Comment by Jint-lzxy** [comment]
Not quite sure what u mean - For 99% of the time `nvim` doesn't ask for elevated permissions, it'll read the login user's config from `~/.config/nvim`. `:h config`:
```console
7. Load user config (execute Ex commands from files, environment, …).
	$VIMINIT environment variable is read as one Ex command line (separate
	multiple commands with '|' or <NL>).
					*config* *init.vim* *init.lua* *vimrc* *exrc*
	A file containing initialization commands is generically called
	a "vimrc" or config file.  It can be either Vimscript ("init.vim") or
	Lua ("init.lua"), but not both. *E5422*
	See also |vimrc-intro| and |base-directories|.

	The config file is located at:
	Unix			~/.config/nvim/init.vim		(or init.lua)
	Windows			~/AppData/Local/nvim/init.vim	(or init.lua)
	|$XDG_CONFIG_HOME|	$XDG_CONFIG_HOME/nvim/init.vim	(or init.lua)

	<...>
						*system-vimrc* *sysinit.vim*
     a. The system vimrc file is read for initializations.  If
	nvim/sysinit.vim file exists in one of $XDG_CONFIG_DIRS, it will be
	used.  Otherwise the system vimrc file is used. The path of this file
	is given by the |:version| command.  Usually it's "$VIM/sysinit.vim".

						*VIMINIT* *EXINIT* *$MYVIMRC*
     b. Locations searched for initializations, in order of preference:
	-  $VIMINIT environment variable (Ex command line).
	-  User |config|: $XDG_CONFIG_HOME/nvim/init.vim (or init.lua).
	-  Other config: {dir}/nvim/init.vim (or init.lua) where {dir} is any
	   directory in $XDG_CONFIG_DIRS.
	-  $EXINIT environment variable (Ex command line).
	|$MYVIMRC| is set to the first valid location unless it was already
	set or when using $VIMINIT.
```
To put it the other way, just copy ur config to `~/.config/nvim/`, which is also what our install script does for u.

**Reply by charliie-dev** [reply]
i'm 98% sure u can put everything under user space, b/c it's my current practice. no root permission is needed at all.

**Comment by Saton-anger** [comment]
Thanks! I will try that.

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: Charles Chiu ***@***.***>
发送时间: Wednesday, August 23, 2023 7:46:38 PM
收件人: ayamir/nvimdots ***@***.***>
抄送: Xu, Zhoutong ***@***.***>; Author ***@***.***>
主题: Re: [ayamir/nvimdots] Upload local neovim systems to remote server. (Discussion #967)

你通常不会收到来自 ***@***.*** 的电子邮件。了解这一点为什么很重要<https://aka.ms/LearnAboutSenderIdentification>

i'm 98% sure u can put everything under user space, b/c it's my current practice. no root permission is needed at all.

―

**Comment by ayamir** [comment]
Just create an archive from local settings and upload it to your server under your home directory. Extract the config to `~/.config/nvim`.
